container/containerd/handler.go

```diff
@@ -91,10 +93,26 @@ func newContainerdContainerHandler(
 		return nil, err
 	}
 
-	taskPid, err := client.TaskPid(ctx, id)
-	if err != nil {
-		return nil, err
+	// Cgroup is created during task creation. When cadvisor sees the cgroup,
+	// task may not be fully created yet. Use a retry+backoff to tolerant the
+	// race condition.
+	// TODO(random-liu): Use cri-containerd client to talk with cri-containerd
+	// instead. cri-containerd has some internal synchronization to make sure
+	// `ContainerStatus` only returns result after `StartContainer` finishes.
+	var taskPid uint32
+	backoff := 100 * time.Millisecond
+	for retry := 5; retry > 0; retry-- {
```

Comment:
so after 5 retries we continue on?  Will we not get any metrics in this case?  Would it be better just to return an error so it doesnt fail silently?

Reply:
... my bad

Reply:
Done